src/adapter/src/coord/command_handler.rs

```diff
@@ -207,6 +207,25 @@ impl Coordinator {
         cancel_tx: Arc<watch::Sender<Canceled>>,
         tx: oneshot::Sender<Response<StartupResponse>>,
     ) {
+        if !session.user().is_superuser() {
+            match self.validate_resource_limit(
+                self.active_conns.len(),
+                i64::try_from(self.active_conns.len() + 1).expect("safe unless the map is too big"),
+                SystemVars::max_connections,
+                "a connection",
+                "max connections",
+            ) {
+                Ok(()) => {}
+                Err(e) => {
+                    let _ = tx.send(Response {
+                        result: Err(e),
+                        session,
+                    });
+                    return;
+                }
+            }
```

Comment:
no need to make this change, you could also rewrite this with a `if let`, e.g.
```
if let Err(e) = self.validate_resource_limit(...) {
  let _ = tx.send(...);
  return;
}
```
this does nothing if the result is `Ok(...)` like we're currently doing

Reply:
done